singularity/code/i18n.py

```diff
@@ -39,12 +41,27 @@
 #It is required that default language have all data files and all of them
 # must have all available entries
 default_language = "en_US"
+
+gettext_language = None
+
+# Prepare main locale dir
+try:
+    main_localedir = dirs.get_writable_file_in_dirs('locale', 'i18n')
```

Comment:
@nthykier Any idea why the key `i18n` doesn't exist when I run `PYTHONPATH=. py.test-3 -v` instead of `./run_singularity.sh`?

Reply:
I think it is because these functions are not generally available until a successful `dirs.create_directories(<...>)` call.  During setup, the initialization is probably at lot more "careful" about bootstrapping order than the tests are.

I think the proper solution here is to wrap that into a `def _get_main_localedir` that lazily computes the directory on the first call (and can return a cached value from there on).

Reply:
Where would this go in the code to be a proper fix?

I tested the following:

1. Change the parameter in `create_directories(True)` in `test_translations.py` to `False` for the module setup. Does not work.
2. Change the parameter in `create_directories(True)` in `test_translations.py` to `g.force_single_dir` for the module setup. Does not work.
3. Add `dirs.create_directories(g.force_single_dir)` to `i18n.py`. This will fix the tests, but list every savegame twice on the "Load Game" screen, which makes this idea unusable.

Reply:
Could you please try with the following method (instead of the `create_directories(...)` approach that you are trying now)

```python
main_localedir = None

def _get_main_localedir():
    global main_localedir
    if main_localedir is None:
        main_localedir = dirs.get_writable_file_in_dirs('locale', 'i18n')
    return main_localedir

[...]

    gettext_language = gettext.translation(TEXTDOMAIN_PREFIX + 'messages', _get_main_localedir(), languages=[lang], fallback=True)
```

Reply:
Thanks, that fixed it. This PR is now ready for review :)